.github/workflows/build_and_test.yml

```diff
@@ -394,6 +394,31 @@ jobs:
         ./build/mvn $MAVEN_CLI_OPTS -DskipTests -Pyarn -Pmesos -Pkubernetes -Phive -Phive-thriftserver -Phadoop-cloud -Djava.version=11 install
         rm -rf ~/.m2/repository/org/apache/spark
 
+  java-17:
```

Comment:
@dongjoon-hyun does this look correct?
I think we're not quite on to even testing vs Java 14 yet, though informally it seems to work fine already.
We may not need to set this up yet, as I imagine anyone wanting to enable this is building from scratch anyway (we won't publish releases requiring 17 for a long time!)

Reply:
Happy to remove it if you don't think it has a place in this PR. I added it simply to get some tests ran on Spark's CI with Java 17, but not in the optic of pushing Spark to test more extensively on the latest version of the JDK.

Reply:
I think we can remove it ... just because I'm not sure it'll even pass although it's very close from some manual testing. We'll probably add Java 14 as a profile soon, which is all but the same for test purposes as 17 (w.r.t Spark) and eventually yes Java 17. I don't feel strongly about it